src/common/cockpitchannel.c

```diff
@@ -1089,14 +1089,17 @@ cockpit_channel_control (CockpitChannel *self,
    * and let close send the message */
   else if (g_str_equal (command, "close"))
     {
-      if (!priv->close_options)
+      if (options)
         {
-          /* Ref for close_options, freed in parent */
-          priv->close_options = json_object_ref (options);
-        }
+          if (!priv->close_options)
+            {
+              /* Ref for close_options, freed in parent */
+              priv->close_options = json_object_ref (options);
+            }
 
-      if (!cockpit_json_get_string (options, "problem", NULL, &problem))
-        problem = NULL;
+          if (!cockpit_json_get_string (options, "problem", NULL, &problem))
+            problem = NULL;
```

Comment:
If you want: you don't need the fallback case anymore, since you set this above.